pyproject.toml

```diff
@@ -1,7 +1,55 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-# Testing tools configuration
+[tool.poetry]
+# Charm is not packed as a standard Python package; this information is not used
+name = "unused"
+version = "0.1.0"
+description = ""
+authors = []
+
+[tool.poetry.dependencies]
+python = "^3.10"
+ops = "^2.2.0"
+tenacity = "^8.2.2"
+cryptography = "^40.0.2"
+jsonschema = "^4.17.3"
+boto3 = "^1.26.113"
+cosl = "^0.0.5"
+
+[tool.poetry.group.format]
+optional = true
+
+[tool.poetry.group.format.dependencies]
+black = "^23.3.0"
+isort = "^5.12.0"
+
+[tool.poetry.group.lint]
+optional = true
+
+[tool.poetry.group.lint.dependencies]
+black = "^23.3.0"
+isort = "^5.12.0"
+flake8 = "^6.0.0"
```

Comment:
since we are refactoring this, wouldn't be nice time to move to `ruff`

Reply:
I would not mix those topics:
1) migration to `poetry`
2) add `renovate`
3) `ruff` lint

Step-by-step... we are in step 1 here. :-D